lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py

```diff
@@ -936,7 +935,147 @@ def test_check_acknowledgement(self):
+        ret_val = self.srs.check_variable_attributes(self.good_dataset)
 
+        for result in ret_val:
+            self.assertTrue(result.value)
+
+        ret_val = self.srs.check_variable_attributes(self.bad_dataset)
+
+        # test if at least one variable attribute is empty
+        glob_result = True
+        for result in ret_val:
+            if result.value is False:
+                glob_result = False
+
+        self.assertFalse(glob_result)
+
+    def test_check_coordinate_variables(self):
+        self.srs.setup(self.good_dataset)
+        self.assertEqual(len(self.srs.imos_1_3_check._coordinate_variables), 3)
+        ret_val = self.srs.check_coordinate_variables(self.good_dataset)
+        for result in ret_val:
+            self.assertTrue(result.value)
+
+        self.srs.setup(self.bad_dataset)
+        self.assertEqual(len(self.srs.imos_1_3_check._coordinate_variables), 4)
+
+    def test_check_time_variable(self):
+        ret_val = self.srs.check_time_variable(self.good_dataset)
+
+        for result in ret_val:
+            self.assertTrue(result.value)
+
+        ret_val = self.srs.check_time_variable(self.bad_dataset)
+
+        for result in ret_val[1:]:
+            self.assertFalse(result.value)
+
+        self.assertEqual(len(ret_val), 5)
+
+    def test_check_variable_attribute_type(self):
+        ret_val = self.srs.check_variable_attribute_type(self.good_dataset)
+
+        for result in ret_val:
+            self.assertTrue(result.value)
+
+        ret_val = self.srs.check_variable_attribute_type(self.bad_dataset)
+
+        for result in ret_val:
+            if result.value == False: # check l2p_flags:valid_min
+                self.assertFalse(result.value)
+    def test_check_data_variable_present(self):
+        self.srs.setup(self.good_dataset)
+        ret_val = self.srs.check_data_variable_present(self.good_dataset)
+        self.assertEqual(len(ret_val), 1)
+        self.assertTrue(ret_val[0].value)
+
+    def test_check_data_variables(self):
+        self.srs.setup(self.good_dataset)
+        ret_val    = self.srs.check_data_variables(self.good_dataset)
+        passed_var = [r.name[1] for r in ret_val if r.value]
+        self.assertEqual(len(ret_val), 14)
+        self.assertEqual(len(passed_var), 14)
+
+        self.srs.setup(self.bad_dataset)
+        ret_val    = self.srs.check_data_variables(self.bad_dataset)
+        failed_var = [r.name[1] for r in ret_val if not r.value]
+        self.assertEqual(len(ret_val), 15)
+        self.assertEqual(len(failed_var), 1)
+        self.assertEqual(set(failed_var), set(['random_var']))
+
+    @unittest.skipUnless(netCDF4.__netcdf4libversion__ >= '4.3',
+                         'requires netCDF4 library version >= 4.3')
+    def test_check_fill_value(self):
+        ret_val = self.srs.check_fill_value(self.bad_dataset)
+        failed_var = [r.name[1] for r in ret_val if not r.value]
+        self.assertEqual(failed_var, ['random_var'])
```

Comment:
Your code is failing this test. I think it should be the 'lon' variable, instead of 'random_var'.